src/trace/cltrace.cpp

```diff
@@ -433,6 +730,140 @@ cl_mem clCreateImage(cl_context context, cl_mem_flags flags,
     return ret;
 }
 
+cl_mem clCreateImage2D(cl_context context, cl_mem_flags flags,
+                       const cl_image_format* image_format, size_t image_width,
+                       size_t image_height, size_t image_row_pitch,
+                       void* host_ptr, cl_int* errcode_ret) {
+    PFN_clCreateImage2D(context, flags, image_format, image_width, image_height,
+                        image_row_pitch, host_ptr, errcode_ret);
+
+    cl_image_desc image_desc = {CL_MEM_OBJECT_IMAGE2D,
+                                image_width,
+                                image_height,
+                                1, // image_depth
+                                1, // image_array_size
+                                image_row_pitch};
+
+    return clCreateImage(context, flags, image_format, &image_desc, host_ptr,
+                         errcode_ret);
+}
+
+cl_mem clCreateImage3D(cl_context context, cl_mem_flags flags,
+                       const cl_image_format* image_format, size_t image_width,
+                       size_t image_height, size_t image_depth,
+                       size_t image_row_pitch, size_t image_slice_pitch,
+                       void* host_ptr, cl_int* errcode_ret) {
+    PFN_clCreateImage3D(context, flags, image_format, image_width, image_height,
+                        image_depth, image_row_pitch, image_slice_pitch,
+                        host_ptr, errcode_ret);
+
+    cl_image_desc image_desc = {CL_MEM_OBJECT_IMAGE3D,
+                                image_width,
+                                image_height,
+                                image_depth,
+                                1, // image_array_size
+                                image_row_pitch,
+                                image_slice_pitch};
+
+    return clCreateImage(context, flags, image_format, &image_desc, host_ptr,
+                         errcode_ret);
+}
+
```

Comment:
```suggestion
```
These look incorrect. You're correctly calling the implementation for clCreateImage2D/3D but then recursing into clCreateImage which will create the image object a second time and capture the wrong API call. I'll delete the implementation for these two API calls before getting the PR in.